frontend/packages/pipelines-plugin/src/components/pipelines/pipeline-metrics/PipelineRunDurationGraph.tsx

```diff
@@ -0,0 +1,84 @@
+import * as React from 'react';
+import { ChartThemeColor, ChartVoronoiContainer } from '@patternfly/react-charts';
+import { LoadingInline } from '@console/internal/components/utils';
+import { GraphEmpty } from '@console/internal/components/graphs/graph-empty';
+import { formatDuration } from '@console/internal/components/utils/datetime';
+import { usePipelineRunDurationPoll } from '../hooks';
+import { getRangeVectorData, PipelineMetricsGraphProps } from './pipeline-metrics-utils';
+import { LineChart } from './charts/lineChart';
+import { DEFAULT_CHART_HEIGHT } from '../const';
+
+import './PipelineChart.scss';
+
+const PipelineRunDurationGraph: React.FC<PipelineMetricsGraphProps> = ({
+  pipeline,
+  timespan,
+  interval,
+  width = 1000,
+}) => {
+  const {
+    metadata: { name, namespace },
+  } = pipeline;
+
+  const [runData, runDataError, runDataLoading] = usePipelineRunDurationPoll({
+    name,
+    namespace,
+    timespan,
+    delay: interval,
+  });
+  const pipelineRunResultData = runData?.data?.result ?? [];
+  if (runDataLoading) {
+    return <LoadingInline />;
+  }
+
+  if (runDataError || pipelineRunResultData.length === 0) {
+    return <GraphEmpty height={DEFAULT_CHART_HEIGHT} />;
+  }
+
+  const pRuns = getRangeVectorData(runData, (r) => r?.metric?.pipelinerun) ?? [];
+  const finalArray = pRuns.reduce(
+    (acc, prun) => {
+      if (!prun) return acc;
+      const obj = prun[prun.length - 1];
+      acc.totalDuration += obj.y;
+      acc.duration.push({ ...obj, time: formatDuration(obj.y * 1000) });
+      return acc;
+    },
+    { totalDuration: 0, duration: [] },
+  );
+  const averageDuration = finalArray.totalDuration
+    ? formatDuration((finalArray.totalDuration * 1000) / finalArray.duration.length)
+    : 0;
+
+  return (
+    <div className="row pipeline-double-chart-container">
+      <div className="col-lg-3 pipeline-run-average-container">
+        <div className="pipeline-run-average-container__duration">{`${averageDuration}`}</div>
+        <div className="pipeline-run-average-container__duration-desc">
+          Average Pipeline Run Duration
+        </div>
+      </div>
+      <div className="col-lg-9">
+        <LineChart
+          ariaDesc="Pipeline Run Duration"
+          ariaTitle="Pipeline Run Duration"
```

Comment:
I believe these need to be translated as well for consistency with screen readers + visible text.